src/Dapper.FluentMap.Dommel/Resolvers/DommelKeyPropertyResolver.cs

```diff
@@ -8,51 +8,49 @@
 namespace Dapper.FluentMap.Dommel.Resolvers
 {
     /// <summary>
-    /// Implements the <see cref="DommelMapper.IKeyPropertyResolver"/> interface by using the configured mapping.
+    /// Implements the <see cref="IKeyPropertyResolver"/> interface by using the configured mapping.
     /// </summary>
-    public class DommelKeyPropertyResolver : DommelMapper.IKeyPropertyResolver
+    public class DommelKeyPropertyResolver : IKeyPropertyResolver
     {
-        /// <inheritdoc/>
-        public PropertyInfo ResolveKeyProperty(Type type)
-        {
-            bool isIdentity;
-            return ResolveKeyProperty(type, out isIdentity);
-        }
+        private readonly IKeyPropertyResolver DefaultResolver = new DefaultKeyPropertyResolver();
 
         /// <inheritdoc/>
-        public PropertyInfo ResolveKeyProperty(Type type, out bool isIdentity)
+        public ColumnPropertyInfo[] ResolveKeyProperties(Type type)
         {
             IEntityMap entityMap;
             if (!FluentMapper.EntityMaps.TryGetValue(type, out entityMap))
             {
-                return DommelMapper.Resolvers.Default.KeyPropertyResolver.ResolveKeyProperty(type, out isIdentity);
+                return DefaultResolver.ResolveKeyProperties(type);
             }
 
             var mapping = entityMap as IDommelEntityMap;
             if (mapping != null)
             {
-                var keyPropertyMaps = entityMap.PropertyMaps.OfType<DommelPropertyMap>().Where(e => e.Key).ToList();
+                var allPropertyMaps = entityMap.PropertyMaps.OfType<DommelPropertyMap>();
+                var keyPropertyMaps = allPropertyMaps.Where(e => e.Key);
+                var keyPropertyInfos = keyPropertyMaps.Select(x => new ColumnPropertyInfo(x.PropertyInfo, isKey: true)).ToArray();
```

Comment:
```suggestion
                var keyPropertyInfos = allPropertyMaps
                     .Where(e => e.Key)
                     .Select(x => new ColumnPropertyInfo(x.PropertyInfo, isKey: true))
                     .ToArray();
```

Combine the LINQ methods into one variable declaration.